includes/widgets/common.php

```diff
@@ -981,6 +981,461 @@ protected function register_controls() {
 			]
 		);
 
+		$this->start_controls_tabs( '_tabs_positioning' );
+
+		foreach ( [ 'normal', 'hover' ] as $tab ) {
+			$this->start_controls_tab(
+				"_tab_positioning_{$tab}",
+				[
+					'label' => 'normal' === $tab ? __( 'Normal', 'elementor' ) : __( 'Hover', 'elementor' ),
+				]
+			);
+
+			$this->add_control(
+				"_{$tab}_transform_rotate_popover",
+				[
+					'label' => __( 'Rotate', 'elementor' ),
+					'type' => Controls_Manager::POPOVER_TOGGLE,
+					'prefix_class' => 'e-',
+					'return_value' => 'transform',
+				]
+			);
+
+			$this->start_popover();
+
+			$this->add_responsive_control(
+				"_{$tab}_transform_rotateZ_effect",
+				[
+					'label' => __( 'Rotate', 'elementor' ),
+					'type' => Controls_Manager::SLIDER,
+					'range' => [
+						'px' => [
+							'min' => -360,
+							'max' => 360,
+						],
+					],
+					'default' => [
+						'size' => 0,
+					],
+					'selectors' => [
+						'{{WRAPPER}} > .elementor-widget-container' => "--{$tab}-rotate-z: {{SIZE}}deg",
```

Comment:
1. Change the variables names to start with --e-transform-xxx (change everywhere).
2. Add a function that gets the $tab variable and return either an empty string or ':hover'.

Reply:
1. Done
2. I think I did what you mean, if not let me know

Reply:
2. Confirmed.